src/addin/addin-client.ts

```diff
@@ -288,25 +296,33 @@ export class AddinClient {
   }
 
   /**
+   * Checks if the height of the iFrame has changed since it was last
+   * posted to the host page (or if it hasn't been posted yet) and initiates
+   * a new post if so.
+   */
+  private checkForHeightChangesOfAddinContent() {
+    const style = getComputedStyle(document.body);
+    const newHeight = document.body.offsetHeight + parseInt(style.marginTop, 10) + parseInt(style.marginBottom, 10);
+
+    if (newHeight !== this.lastPostedIframeHeight) {
+      this.lastPostedIframeHeight = newHeight;
+
+      this.postMessageToHostPage({
+        message: {
+          height: newHeight + 'px'
+        },
+        messageType: 'height-change'
+      });
+    }
+  }
+
+  /**
    * Starts a timeout interval to watch for height changes
    * of the iframe content.
    */
   private trackHeightChangesOfAddinContent() {
-    let currentHeight: number;
-
     this.heightChangeIntervalId = setInterval(() => {
-      const newHeight = document.body.offsetHeight;
-
-      if (newHeight !== currentHeight) {
-        currentHeight = newHeight;
-
-        this.postMessageToHostPage({
-          message: {
-            height: newHeight + 'px'
-          },
-          messageType: 'height-change'
-        });
-      }
+      this.checkForHeightChangesOfAddinContent();
```

Comment:
Could this be simplified to the following?

```
this.heightChangeIntervalId = setInterval(this.checkForHeightChangesOfAddinContent, 1000);
```

Reply:
No, it can't b/c it will lose it's "this" (scope) context.  Wrapping in () => preserves it.

Reply:
Go home JS.  You're drunk.  Good catch.  Could use apply or call I suppose, but I imagine it's more readable as you have it now.